### PR TITLE
[rush] Fix annoyance where "rush generate" churns tarball hashes in shrinkwrap.yaml

### PR DESCRIPTION
**Pull request by pgonzal**
The PNPM **shrinkwrap.yaml** file has a bunch of entries like this:

```yaml
    resolution:
      integrity: sha512-uQ0huPs3icbmDjSQSmg6RCND2sHu9Afw+GxEDw2fKUjoAoDJC8HacQsQ49FDCbcP8WATZoACcXr/dezjUkiWMg==
      tarball: 'file:projects/loader-set-webpack-public-path.tgz'
```

People found it annoying that every PR with "rush generate" would have a diff for these lines, even though nothing changed about those packages.  The reason was that file timestamps were affecting this hash.

### CODE REVIEW

File: apps/rush-lib/src/cli/logic/InstallManager.ts [hunk 1]
```diff
@@ -34,6 +34,19 @@ const MAX_INSTALL_ATTEMPTS: number = 5;
 
 const wrap: (textToWrap: string) => string = wordwrap.soft(Utilities.getConsoleWidth());
 
+/**
+ * The "noMtime" flag is new in tar@4.4.1 and not available yet for \@types/tar.
+ * As a temporary workaround, augment the type.
+ */
+import { CreateOptions } from 'tar';
+export interface CreateOptions { // tslint:disable-line:interface-name
```
Comment (iclanton) [inline apps/rush-lib/src/cli/logic/InstallManager.ts]: >CreateOptions [](start = 17, length = 13)

Does this somehow append to the definitions coming from the @types package?

Comment (pgonzal) [reply apps/rush-lib/src/cli/logic/InstallManager.ts]: Yes, it is [declaration merging](https://www.typescriptlang.org/docs/handbook/declaration-merging.html)

---
In reply to: [178998630](https://github.com/Microsoft/web-build-tools/pull/595#discussion_r178998630) [](ancestors = 178998630)

**Comment by HipsterZipster** [comment]
@pgonzal When can we release this ticket? I think I'm facing this issue as well

**Comment by pgonzal** [comment]
It's already been published as part of the beta release `rush@5.0.0-dev.18`.  Unfortunately Rush 5 will have some significant breaking changes, so it will probably bake until the end of this month before we officially release it.

**Comment by pgonzal** [comment]
If you want to dogfood Rush 5 in your repo, let me know and I can explain the tradeoffs and caveats.

**Comment by HipsterZipster** [comment]
I'm definitely happy to try out Rush 5.  Can you give me some pointers?

**Comment by HipsterZipster** [comment]
Using:
Windows 7 (corporate environment)
Node 8.11.1 LTS
"@microsoft/rush": "5.0.0-dev.20",
"pnpmVersion": "1.43.1",

Here's what I see so far:

- The syntax on the `command-line.json` is more strict
- `rush update` is used instead of `rush generate`
- There are some new output messages from the CLI: 

> The NPM shrinkwrap file does not provide "[package-name]" ([package-version]) required by the preferred versions from common-versions.json

This log message doesn't seem to make much sense to me since I don't have a `common-versions.json` file? What is that?

**Pull request merged**